dissect/target/filesystems/btrfs.py

```diff
@@ -181,6 +181,9 @@ def lstat(self) -> fsutil.stat_result:
 
         # Add block information of the filesystem
         st_info.st_blksize = entry.btrfs.sector_size
-        st_info.st_blocks = math.ceil(entry.size / st_info.st_blksize)
+
+        st_info.st_blocks = 0
+        if not self.is_dir():
+            st_info.st_blocks = math.ceil(st_info.st_blksize / 512) * math.ceil(st_info.st_size / st_info.st_blksize)
```

Comment:
I'm not sure this double ceiling is correct, it could lead to a really inflated number of blocks, e.g.:

```
st_size: 256
st_blksize: 128
->
math.ceil(128/512) * math.ceil(256/128) = math.ceil(1/4) * math.ceil(2) = 1 * 2 = 2
```
Inflates the apparent size by a factor of 4

```
st_size: 1024
st_blksize: 768
->
math.ceil(768/512) * math.ceil(1024/768) = math.ceil(3/2) * math.ceil(4/3) = 2 * 2 = 4
```
Inflates the apparent size by a factor of 2

If I read and interpret the various sources of what `st_blocks` should be it seems to me it represents the size *on disk* in blocks (which are 512 bytes in our implementation).
The POSIX/opengroup for example says: "Number of blocks allocated for this object."

The on disk is important in the sense that if files that are e.g. sparse their size on disk is smaller than their actual size, the same holds for example for files that are compressed on the filesystem layer.

So I think we can just do a `math.ceil(st_info.st_size / 512)` here. It also means that for symbolic links `st_blocks` should be 0, unless the symbolic link data is not (just) contained in the inode structure.